### PR TITLE
Add Mecanum vehicle example

### PR DESCRIPTION
**Pull request by MarqRazz**
As the title says this adds  a new example showing how to use the new `MecanumDriveController` in ros2_controllers :trophy:.
mecanum_vehicle

The odometry of the vehicle is still a little off when rotating because I don't fully understand what [sum_of_robot_center_projection_on_X_Y_axis](https://github.com/ros-controls/ros2_controllers/blob/master/mecanum_drive_controller/src/mecanum_drive_controller.yaml#L109-L114) parameter would equate to for this vehicle.

### CODE REVIEW

**Comment by MarqRazz** [comment]
Looks like this might be blocked on the mecanum controller getting released.
`gz_ros2_control_demos: Cannot locate rosdep definition for [mecanum_drive_controller]`

**Comment by christophfroehlich** [comment]
@bmagyar just released it ;)

**Comment by bmagyar** [comment]
I am Santa, reading your wishlists on github

https://github.com/ros/rosdistro/pull/43821
https://github.com/ros/rosdistro/pull/43822

File: gz_ros2_control_demos/launch/mecanum_drive_example.launch.py
Comment (ahcorde) [inline gz_ros2_control_demos/launch/mecanum_drive_example.launch.py]:
```suggestion
# Copyright 2024 Open Source Robotics Foundation, Inc.
```

**Review by ahcorde — APPROVED** [review]
Just a minor comment, otherwise lgtm when the ci is green. I just merged the PRs in ros_distro

**Comment by ahcorde** [comment]
https://github.com/Mergifyio backport jazzy

**Comment by ahcorde** [comment]
@Pratham-Pandey can you help us to setup `sum_of_robot_center_projection_on_X_Y_axis` ?

**Comment by Pratham-Pandey** [comment]
From what I understood,  `sum_of_robot_center_projection_on_X_Y_axis` is the sum of distance between robot frame(usually at the center) and the wheels. For example if the robot have 4 wheels, we will consider distance to any one wheel(assuming each wheel is equidistant from the robot frame). The distance in x axis plus the distance in y axis.

**Comment by MarqRazz** [comment]
Thanks for the explanation @Pratham-Pandey! For some reason I half'd the value to the wheels in the X and Y thinking I had that as the total wheel base. This is corrected now and reports odom that matches the pose of the vehicle in Gazebo.

**Comment by ahcorde** [comment]
https://github.com/Mergifyio backport jazzy

**Comment by soham2560** [comment]
any chance this will be ported to humble?

**Comment by christophfroehlich** [comment]
not before https://github.com/ros-controls/ros2_controllers/pull/1376 is fixed

**Comment by soham2560** [comment]
@christophfroehlich now that https://github.com/ros-controls/ros2_controllers/pull/1376 has been merged, can we start of with the humble backport?

**Comment by christophfroehlich** [comment]
@soham2560 see #515

**Comment by christophfroehlich** [comment]
@mergifyio backport rename/ign/gz

**Pull request merged**
